File: threads02/csharp/ex02/View/ResultadosView.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ex02.View;

public class ResultadosView
{
    public void MostrarResultado(List<string> nomesOriginais, List<List<string>> partes, List<string> resultado)
    {
        Console.WriteLine("----- EXERCÍCIO 2: FILTRO DE DADOS -----");

        Console.WriteLine($"Total de nomes originais: {nomesOriginais.Count}");
        Console.WriteLine($"Quantidade de tarefas: {partes.Count}");

        for (int i = 0; i < partes.Count; i++)
        {
            Console.WriteLine($"Parte {i + 1}: {partes[i].Count} nomes");
        }

        Console.WriteLine($"Total processado: {resultado.Count}");
        Console.WriteLine("--------------------------------------------------");
        Console.WriteLine("Exemplos após limpeza:\n");

        foreach (string nome in resultado.Take(10))
        {
            Console.WriteLine($" - {nome}");
        }

        Console.WriteLine("--------------------------------------------------");
    }
}

File: 1.2-threads/java/ex02/src/view/ResultadosView.java
package view;

import java.util.List;

public class ResultadosView {

    /**
     * mostra o resultado do processamento
     * @param nomesOriginais lista de nomes originais
     * @param partes lista de partes
     * @param resultado lista de resultados
     */
    public void mostrarResultado(List<String> nomesOriginais, List<List<String>> partes, List<String> resultado) {

        System.out.println("----- EXERCÍCIO 2: FILTRO DE DADOS -----");
        
        System.out.println("Total de nomes originais: " + nomesOriginais.size());
        System.out.println("Quantidade de tarefas: " + partes.size());

        for (int i = 0; i < partes.size(); i++) {
            System.out.println("Parte " + (i + 1) + ": " + partes.get(i).size() + " nomes");
        }

        System.out.println("Total processado: " + resultado.size());
        System.out.println("--------------------------------------------------");

        System.out.println("Exemplos após limpeza:\n");

        resultado.stream().limit(10).forEach(nome -> System.out.println(" - " + nome));

        System.out.println("--------------------------------------------------");
    }
}
